xarray/backends/lru_cache.py

```diff
@@ -0,0 +1,89 @@
+import collections
+import threading
+
+
+class LRUCache(collections.MutableMapping):
+    """Thread-safe LRUCache based on an OrderedDict.
+
+    All dict operations (__getitem__, __setitem__, __contains__) update the
+    priority of the relevant key and take O(1) time. The dict is iterated over
+    in order from the oldest to newest key, which means that a complete pass
+    over the dict should not affect the order of any entries.
+
+    When a new item is set and the maximum size of the cache is exceeded, the
+    oldest item is dropped and called with ``on_evict(key, value)``.
+
+    The ``maxsize`` property can be used to view or adjust the capacity of
+    the cache, e.g., ``cache.maxsize = new_size``.
+    """
+    def __init__(self, maxsize, on_evict=None):
+        """
+        Parameters
+        ----------
+        maxsize : int
+            Integer maximum number of items to hold in the cache.
+        on_evict: callable, optional
+            Function to call like ``on_evict(key, value)`` when items are
+            evicted.
+        """
+        if maxsize < 0:
+            raise ValueError('maxsize must be non-negative')
+        self._maxsize = maxsize
```

Comment:
Should we enforce maxsize is an integer? I'm thinking that it may be easy to see `None`/`False` as valid values. I think that case is going to break things downstream.

Reply:
done